### DISCUSSION TITLE
"Co-Authored by" Commit Problem

### DISCUSSION THREAD

**Discussion by kaplanark** [community/community#36076] [Pull Requests] [answered]
Hi, we were going to do Pull Request with Co-Author in Github, but my friend did not appear in Commit. I did it as it was written in documentation, but it didn't. My friend's name appeared only as a flat message. Why is this caused by? Why couldn't I do it?

I did it like this but it didn't work:

`$ git commit -m "Refactor usability tests.`
`>`
`>`
`Co-authored-by: NAME [NAME@EXAMPLE.COM](mailto:NAME@EXAMPLE.COM)`
`Co-authored-by: AUTHOR-NAME [ANOTHER-NAME@EXAMPLE.COM](mailto:ANOTHER-NAME@EXAMPLE.COM)"`

**Comment by devisasari** [comment, marked as answer]
This method sometimes causes problems, but I recommend Github Desktop as a smooth method. There you can easily do it by adding the user name to commit. Look like that:

<img width="190" alt="deneme" src="https://user-images.githubusercontent.com/99514353/195724253-4e977bbc-2a35-4405-8c56-7953c6149beb.png">

**Comment by brianzelip** [comment]
FWIW, my problem, on a multi-line commit message, was solved by adding one blank line before the final co-author line:

```sh
# problem
> git commit -m "Refactor interface

Add translations
Add npm scripts
Co-authored-by XXX <xxx@example.com>"
```

```sh
# solution
> git commit -m "Refactor interface

Add translations
Add npm scripts

Co-authored-by XXX <xxx@example.com>"
```

**Reply by profnandaa** [reply]
@brianzelip -- any tips on how to do it when you don't have the user's email address coz of GH's privacy protection?

**Reply by airtower-luna** [reply]
Ask them to give you their [GitHub-provided "noreply" pseudo-address](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-commit-email-addresses), or look it up from one of their previous commits. It'll work just as well. :slightly_smiling_face:

**Reply by brianzelip** [reply]
@profnandaa what @airtower-luna said!
